src/server/api/routers/event.ts

```diff
@@ -80,4 +85,20 @@ export const eventRouter = createTRPCRouter({
       throw e;
     }
   }),
+  byName: publicProcedure.input(byNameSchema).query( async ({input, ctx}) => {
+    const {name, sortByDate} = input; 
+
+    try { 
+      const events = await ctx.db.query.events.findMany({
+        where: (event) => ilike(event.name, `%${name}%`),
```

Comment:
We should maybe filter out events that have already ended, but I'll leave that decision to the next reviewer

Reply:
I think it should prioritize upcoming events but for now we can keep ended events just in case people want to look at older events

Reply:
Yeah, that makes sense, at some point we should maybe set a limit for how far back it would show but that's definitely out of scope of this pr.